cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ApplicationSpecificationAdapter.java

```diff
@@ -96,7 +102,10 @@ public static GsonBuilder addTypeAdapters(GsonBuilder builder) {
       .registerTypeAdapter(SatisfiableTrigger.class, new TriggerCodec())
       .registerTypeAdapter(Constraint.class, new ConstraintCodec())
       .registerTypeAdapterFactory(new AppSpecTypeAdapterFactory())
-      .registerTypeAdapter(Requirements.class, new RequirementsCodec());
+      .registerTypeAdapter(Requirements.class, new RequirementsCodec())
+      .registerTypeAdapter(TriggerInfo.class, new TriggerInfoCodec())
```

Comment:
It doesn't look like these are being added to the ApplicationSpecification, so they shouldn't be here in the adapter. Instead, register the codecs with the GsonBuilder directly in the class you need it.

Though I'm also not clear why we need the codecs if we're adding a bunch of proto classes?